test/parallel_api/iterator/input_data_sweep.pass.cpp

```diff
@@ -0,0 +1,348 @@
+// are being tested, and to limit the number of types to be within reason.
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <int __recurse, int __reverses, bool __read = true, bool __reset_read = true, bool __write = true,
+          bool __check_write = true, bool __usable_as_perm_map = true, bool __usable_as_perm_src = true,
+          bool __is_reversible = true, typename Policy, typename InputIterator1, typename InputIterator2,
+          typename OutputIterator, typename OriginalIterator1, typename OriginalIterator2, typename ExpectedIterator,
+          typename T>
+void
+wrap_recurse(Policy&& exec, InputIterator1 first, InputIterator1 last, InputIterator2 copy_from_first,
+             OutputIterator copy_to_first, OriginalIterator1 orig_first, OriginalIterator2 orig_out_first,
+             ExpectedIterator expected_first, T trash, std::string input_descr)
+{
+    auto exec1 = TestUtils::create_new_policy_idx<Policy, 0>(exec);
+    auto exec2 = TestUtils::create_new_policy_idx<Policy, 1>(exec);
+    auto exec3 = TestUtils::create_new_policy_idx<Policy, 2>(exec);
+    auto exec4 = TestUtils::create_new_policy_idx<Policy, 3>(exec);
+    auto exec5 = TestUtils::create_new_policy_idx<Policy, 4>(exec);
+    auto exec6 = TestUtils::create_new_policy_idx<Policy, 5>(exec);
+    auto exec7 = TestUtils::create_new_policy_idx<Policy, 6>(exec);
+    auto exec8 = TestUtils::create_new_policy_idx<Policy, 7>(exec);
+    auto exec9 = TestUtils::create_new_policy_idx<Policy, 8>(exec);
+    auto exec10 = TestUtils::create_new_policy_idx<Policy, 9>(exec);
+    auto exec11 = TestUtils::create_new_policy_idx<Policy, 10>(exec);
+    auto exec12 = TestUtils::create_new_policy_idx<Policy, 11>(exec);
+    auto exec13 = TestUtils::create_new_policy_idx<Policy, 12>(exec);
+
+    oneapi::dpl::counting_iterator<size_t> counting(size_t{0});
+
+    const auto n = last - first;
+
+    //Run the tests
+    auto get_expect = [n](auto exp) {
+        if constexpr (__reverses % 2 == 0)
+        {
+            return exp;
+        }
+        else
+        {
+            return std::make_reverse_iterator(exp + n);
+        }
+    };
+
+#    if VERBOSE_TEST
+    std::cout << input_descr << ":";
+#    endif // VERBOSE_TEST
+
+    if constexpr (__read)
+    {
+        oneapi::dpl::fill(exec1, orig_out_first, orig_out_first + n, trash);
+        if constexpr (__reset_read)
+        {
+            //Reset data if required
+            oneapi::dpl::copy(exec2, counting, counting + n, orig_first);
+        }
+
+        //Run test
+        oneapi::dpl::copy(exec3, first, last, copy_to_first);
+
+        //get expected sequence with proper number of reverses
+        auto expect = get_expect(expected_first);
+        std::string msg = std::string("wrong read effect from ") + input_descr;
+        //verify result using original unwrapped output
+        EXPECT_EQ_N(expect, orig_out_first, n, msg.c_str());
+#    if VERBOSE_TEST
```

Comment:
Probably you may use `_ONEDPL_DEBUG_SYCL` instead of this define.

Reply:
done

Reply:
ok